src/data/restore.ts

```diff
@@ -46,8 +47,12 @@ export function restore(
         } else {
           points = element.points;
         }
+        element.points = points;
       } else {
         normalizeDimensions(element);
+        if ("points" in element) {
+          delete element.points;
```

Comment:
Why are you mutating things from elements directly?

Reply:
IMO it doesn't matter for `restore()`. The only way we're using `restore()` other than during scene load is in multiplayer on `SCENE_UPDATE`, and I would remove that call because it doesn't make sense to use it there.

I can switch to non-mutation, but IMO we'd not gain much benefit here. We'd either need to spread twice, or do something like:

```js
let points: Point[] | null = null;
if (element.type === "arrow") {
  /* ... */
}
return {
  ...element,
  ...(points ? { points } : {})
```

And do this for every type-specific property in the future.

Reply:
Actually, the above wouldn't work. The idea is to remove the `points` from non-linear elements, and the above wouldn't do it. So I'd indeed need to do 2 rounds of copying at minimum.

Reply:
Also, if you're worried that mutating the element will result in regressions later. There are only two cases that come to mind:

1. We use `restore()` incorrectly, and mutating won't trigger update? No. Because we're spreading the mutated `element` into a new object, so we'll always get a new reference.
2. We use `restore()` on elements that do not come from a serialized list of elements, which will cause the passed-in elements to be mutated. True. Likelihood of this is low, and in this case it's also not mutating the object in any way that would hurt.

Reply:
I actually didn't put much thoughts into it, I just found it interesting that we used direct mutation here. Your rationale makes sense, it may be worth adding a line or two of comments above to explain.